### DISCUSSION TITLE
Question: Can I use this on a Pivot Model?

### DISCUSSION THREAD

**Discussion by ferdiebergado** [GeneaLabs/laravel-model-caching#485] [General]
This is actually just a question.

I have created a Base Pivot Model which use laravel-model-caching like this:

```php
<?php

namespace App;

use Illuminate\Database\Eloquent\Relations\Pivot;
use GeneaLabs\LaravelModelCaching\Traits\Cachable;

abstract class BasePivot extends Pivot
{
    use Cachable;
}

```

Then, from it I extend my Pivot Models.

```php
<?php

namespace App;

use App\BasePivot;

class UserPermission extends BasePivot
{
    //
}
```

It seems that every time the Pivot Model is called in a relationship, data is not cached as seen from debugbar.

```php
    public function permissions()
    {
             return $this->belongsToMany('App\Permission')->using('App\UserPermission');
    }
```

What am I missing? Thanks.

**Comment by mikebronner** [comment]
Hi @ferdiebergado, I'll have to look into this.

**Comment by ferdiebergado** [comment]
Thanks mike. I'll be standing by for updates.

**Comment by mikebronner** [comment]
This is related to #127 .... so I won't have an answer for you anytime soon. But I'll keep this open. If anyone has the time to create a PR for this, that would be awesome!

**Comment by ferdiebergado** [comment]
Alright mike.  I'll just use eager-loading for relationships for now. Thanks.

**Comment by mikebronner** [comment]
@ferdiebergado Hi there, I know it's been a while, but I might have good news. :) Would you mind testing the latest release and see if this works for you now?
